<launch>
  <arg name="launch_prefix" default="" /> <!-- set to value="gdbserver localhost:10000" for remote debugging -->
  <arg name="node_namespace" default="apriltag_ros_continuous_node" />
  <arg name="camera_name" default="/camera/rgb" />
  <arg name="camera_info" default="raspicam_node" />
  <arg name="camera_frame" default="camera" />
  <arg name="image_topic" default="image_raw" />
  
  
  <!-- Set parameters -->
  <rosparam command="load" file="$(find assignment6_trackingandfollowing)/src/launch/settings.yaml" ns="$(arg node_namespace)" />
  <rosparam command="load" file="$(find assignment6_trackingandfollowing)/src/launch/tags.yaml" ns="$(arg node_namespace)" />
  
  <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="$(arg node_namespace)" clear_params="true" output="screen" launch-prefix="$(arg launch_prefix)" >
    <!-- Remap topics from those used in code to those on the ROS network -->
    <remap from="image_rect" to="$(arg camera_name)/$(arg image_topic)" />
    <remap from="camera_info" to="$(arg camera_info)/camera_info" />

    <param name="camera_frame" type="str" value="$(arg camera_frame)" />
    <param name="publish_tag_detections_image" type="bool" value="true" />      <!-- default: false -->
  </node>
  
  <node pkg="project_turtlebot_maneouvres" type="apriltag_follower.py" name="AprilTag_Following" />
</launch>
